framework/mesh/mesh.cc

```diff
@@ -12,6 +12,9 @@ namespace opensn
 std::shared_ptr<MeshContinuum>
 GetCurrentMesh()
 {
+  if (mesh_stack.empty())
+    throw std::logic_error(
+      "Empty mesh stack. Did you forget \"MeshGenerator.Execute()\" in the input?");
```

Comment:
```suggestion
  OpenSnLogicalErrorIf(mesh_stack.empty(), "Empty mesh stack. Ensure that \"mesh.MeshGenerator.Execute\" is called in the input.");
```
Any reason to not use `OpenSnLogicalErrorIf` here? Are we trying to move away from this currently? Also note the suggested change to the comment method, particularly the inclusion of `mesh.` and the elimination of empty `()` since the routine takes an argument.

Reply:
> Any reason to not use `OpenSnLogicalErrorIf` here? Are we trying to move away from this currently?

AFAIK, no decision has been made. This problem is being tracked by #65, but there was/is no discussion as of today and no plans were made either.

Reply:
I have been thinking about this problem, so I will put some of my recent thoughts there.

Reply:
I 100% agree with all of David's comments in #65. I will sit down and add my own thoughts there as well. I've never been a big fan of printing line numbers and functions names in user errors. This rarely holds any meaning for the user, and, without the corresponding input/data files, isn't much help for developers either. I've been using exceptions whenever possible.

Reply:
I've updated the error message as suggested.